PPM-375 Move the gitlab.com repo used by the scripts

fvbogaert opened the pull request:
We need to move the gitlab repository used by CI from gitlab.com/prpl-foundation/prplMesh to gitlab.com/prpl-foundation/prplmesh/prplMesh.

This PR is for updating the scripts that work with the repository location.

Don't merge until the move is complete on gitlab.com.

It relies on the following changes: https://git.prpl.dev/prplmesh/wfa-certification/easymesh_cert/-/merge_requests/25

fvb-mind committed:
PPM-375 Move the gitlab.com repo used by the scripts

Signed-off-by: Frederik Van Bogaert <frederik.vanbogaert@mind.be>

rmelotte commented on tools/functions.sh:
```diff
@@ -7,7 +7,7 @@
 
 # Docker registry which is used in all docker-related scripts
-export DOCKER_REGISTRY=registry.gitlab.com/prpl-foundation/prplmesh/
+export DOCKER_REGISTRY=registry.gitlab.com/prpl-foundation/prplmesh/prplMesh/
```
Even though the project may have capitalized letters, registry names can only be lowercase:
```suggestion
export DOCKER_REGISTRY=registry.gitlab.com/prpl-foundation/prplmesh/prplmesh/
```

rmelotte commented:
I think the only reference left in the repo is the badge in the README:
```
README.md:[![Build Status](https://gitlab.com/prpl-foundation/prplMesh/badges/master/pipeline.svg)](https://gitlab.com/prpl-foundation/prplMesh/pipelines)
```

rmelotte commented:
Unfortunately I can't find a source for it, but most likely the location where Gitlab clones the repo will change as well.
Up to now it was `/builds/prpl-foundation/prplMesh/`, so it's probably going to be `/builds/prpl-foundation/prplmesh/prplMesh/`.

Ideally we shouldn't be depending on this path anyway, but unfortunately https://jira.prplfoundation.org/browse/PPM-208 is still open so we have no choice but to do more changes.

This path also has to be used by... drum roll... the certification repo, so we will have to update it as well (I can take care of that).

ghost commented:
> Unfortunately I can't find a source for it, but most likely the location where Gitlab clones the repo will change as well.
> Up to now it was `/builds/prpl-foundation/prplMesh/`, so it's probably going to be `/builds/prpl-foundation/prplmesh/prplMesh/`.
> 
```
$ git grep prpl-foundation
.gitlab-ci.yml:    GIT_CLONE_PATH: "/builds/prpl-foundation/prplMesh/"
README.md:[![Build Status](https://gitlab.com/prpl-foundation/prplmesh/prplMesh/badges/master/pipeline.svg)](https://gitlab.com/prpl-foundation/prplmesh/prplMesh/pipelines)
ci/certification/generic.yml:    GIT_CLONE_PATH: "/builds/prpl-foundation/prplMesh/"
tools/download_ipk.sh:        GITLAB_BASE_URL="https://gitlab.com/prpl-foundation/prplmesh/prplMesh/-/jobs/artifacts/"
tools/functions.sh:export DOCKER_REGISTRY=registry.gitlab.com/prpl-foundation/prplmesh/prplmesh/
```
The gitlab docs have this to say:
```
Git clone path
Introduced in GitLab Runner 11.10.

GIT_CLONE_PATH allows you to control where you clone your sources. This can have implications if you heavily use big repositories with fork workflow.

Fork workflow from GitLab Runner’s perspective is stored as a separate repository with separate worktree. That means that GitLab Runner cannot optimize the usage of worktrees and you might have to instruct GitLab Runner to use that.

In such cases, ideally you want to make the GitLab Runner executor be used only for the given project and not shared across different projects to make this process more efficient.

The GIT_CLONE_PATH has to be within the $CI_BUILDS_DIR. Currently, it is impossible to pick any path from disk.
```
I don't think we even need `GIT_CLONE_PATH`, we don't seem to use it anywhere? Unless ...

> This path also has to be used by... drum roll... the certification repo, so we will have to update it as well (I can take care of that).

... it's used in the other repos :-)

**EDIT**: But even so, I don't think the repo location necessarily impacts what value we have to use for `GIT_CLONE_PATH`, we can just keep it the same. Which means the other repos should be unaffected

rmelotte committed:
ci: easymesh_cert: take the new prplMesh path into account

Moving the project on Gitlab.com from prpl-foundation/prplMesh to
prpl-foundation/prplmesh/prplMesh will also change the place where
where the prplMesh repo is checked out.

prplMesh will not only be built in a new location, but also has to be
run in the same new location. See:
https://jira.prplfoundation.org/browse/PPM-208

Take the new prplMesh build path into account.

Signed-off-by: Raphaël Mélotte <raphael.melotte@mind.be>

rmelotte commented:
It doesn't have to do with the fact that we use another rrepo, it's used implicitly by Gitlab.
See for example [the docker builds](https://gitlab.com/prpl-foundation/prplMesh/-/jobs/679790035): they are built in `/builds/prpl-foundation/prplMesh/` even though we didn't specify that path ourselves anywhere.

As for why we need a changes:
- The jobs using a docker runner will build prplMesh in docker in a new location: `/builds/prpl-foundation/prplmesh/prplMesh/`.
- test_flows.py also uses a docker runner, so it will use this new path as well, nothing to change here.
- For the shell runners however, it's a different matter:
  By default they build/run in a folder using $CI_BUILDS_DIR, but with additional random string to allow concurrent jobs.
  Because prplMesh is not relocatable, we instructed them to use `/builds/prpl-foundation/prplMesh/` instead.
- The certification repo has to know where prplMesh is deployed, so it has to be updated as well.

EDIT: I added the changes I describe here in a new commit.

ghost commented:
The move was completed. We'll need to merge this PR soon

rmelotte commented:
@mergifyio rebase

rmelotte commented:
The pipeline related to this PR was deleted, I created a new one to check if it works: https://gitlab.com/prpl-foundation/prplmesh/prplMesh/-/pipelines/177175447

rmelotte commented:
The pipeline succeeded, so I will merge this one and rebase the other branches.
